core/network/source_other.go

```diff
@@ -9,25 +9,90 @@ import (
 	"github.com/juju/errors"
 )
 
-type netPackageConfigSource struct{}
+// netAddr implements ConfigSourceAddr based on an address in string form.
+type netAddr struct {
+	addr  string
+	ip    net.IP
+	ipNet *net.IPNet
+}
+
+// NewNetAddr returns a new netAddr reference representing the input IP address
+// string. No error is returned; instead the validity of input is indicated by
+// the return having a populated ip member.
+// TODO (manadart 2021-02-15): This method is exported on account of testing in
+// the api/common package where this logic used to reside and where the actual
+// detection and conversion to params is invoked.
+// The detection should also be relocated here to core/network in order that
+// the export is no longer required.
+func NewNetAddr(a string) *netAddr {
```

Comment:
I get it that this is only used in tests but I believe it is error prone to defer the actual validation to the consumer as passing a netAddr instance around would generally imply (at least to me) that it is a valid value. Looking forward to this being removed in the future.